### ISSUE TITLE
Deselect all notes with right-click

### ISSUE THREAD

**Issue by Sawuare** [LMMS/lmms#5103]

**Comment by BaraMGB** [comment]
In my opinion all short cuts should be consist with the song editor.

**Comment by pbondoer** [comment]
I'll try to have a go at this during the weekend, if that's alright! 😄

**Comment by Spekular** [comment]
Wouldn't this remove the ability to access a context menu with selected notes? Unless you mean that "deselect all notes" would be a context menu item.

A context menu for selected would be useful for eg cut/copy/quantize selected, and perhaps others in the future (flipping? Etc.).

**Comment by pbondoer** [comment]
@Spekular That is what I noticed while exploring the code. I'm not an LMMS user (yet!) as the UX is throwing me off from my years of using FL Studio. I don't know about other DAWs but FL will deselect notes if you enter a new note into the piano roll, or simply right click in a blank area.

I think it's important to weigh how much the context menu brings to the table compared to being able to quickly deselect everything.

Perhaps we could make the gesture happen only if a modifier key is pressed? Though that might defeat the original intent.

I know for example that FL shows a context menu when you do Shift + Right Click + Left Click (in that specific order while holding down all buttons).

![image](https://user-images.githubusercontent.com/1671753/63637086-671f7200-c678-11e9-9d77-106b44b66fb4.png)

@Sawuare could you perhaps clarify what you had envisionned? I'm not sure what to do 😢

**Comment by Spekular** [comment]
@pbondoer yeah, left click to create a new note (followed by right click to delete, if it's unwanted) will deselect, as will ctrl + clicking an empty are (starts and immediately ends a new box select, resulting in no selected notes).

I suppose RMB is already occupied when clicking on notes though, so a context menu

**Comment by Spekular** [comment]
Whoops, accidentally hit "Comment" too soon. TL;DR of what I meant to write: RMB is already in use to delete notes, so a context menu on unmodified RMB is already out of the question. This suggestion is fine as is, imo, feel free to ignore my earlier comment.

**Comment by musikBear** [comment]
I caused this -i did not know that esc deselect already.  imo thats fine.
Afaik Esc is not bound to any song-editor feature, so isent the right way to go to set all deselct in song-editor, when esc is pressed.
Note  / Grouped notes right-click should be preserved for planned context-options.

**Comment by BaraMGB** [comment]
> Afaik Esc is not bound to any song-editor feature, so isent the right way to go to set all deselct in song-editor, when esc is pressed.

in master it is.

**Comment by musikBear** [comment]
> in master it is.

o ke.. Is it more logical, in its bound Master functionality, than as a *general* de-select?
What does it do in Master?

**Comment by Sawuare** [comment]
> @Sawuare could you perhaps clarify what you had envisionned?

@pbondoer Since everyone seems to be fine with
